inc/problem.class.php

```diff
@@ -284,14 +284,39 @@ function post_updateItem($history = 1) {
 
 
    function prepareInputForAdd($input) {
-
       $input =  parent::prepareInputForAdd($input);
+      if ($input === false) {
+         return false;
+      }
+
+      // Manage auto assign
+      $auto_assign_mode = Entity::getUsedConfig('auto_assign_mode', $input['entities_id']);
 
-      if (((isset($input["_users_id_assign"]) && ($input["_users_id_assign"] > 0))
-           || (isset($input["_groups_id_assign"]) && ($input["_groups_id_assign"] > 0))
-           || (isset($input["_suppliers_id_assign"]) && ($input["_suppliers_id_assign"] > 0)))
-          && (in_array($input['status'], $this->getNewStatusArray()))) {
```

Comment:
This code is also found in tickets. several years ago I updated this code in tickets to allow creation of tickets with several actors (the UI allows only one). I think this is meaningful to bring the enhancement here as well.

If accepted, it would be a good idea to factorize this snippet among ticket, change and problem.